mod/cornercutter/ModFeature/SpecialMods/CornercutterChristmas/GiftEffects.cs

```diff
@@ -0,0 +1,255 @@
+            if (killHit.attacker != Player.singlePlayer && killHit.attacker.wielder != Player.singlePlayer) return;
+            LastKillTime = Time.time;
+        }
+
+        // GiftType.BuyingFromShopsWeakensEnemies
+        static void GiveKillCrit(HitInfo info, CritChanceInfo critInfo)
+        {
+            if (info.attacker != Player.singlePlayer || (LastKillTime + 7 * CurrentStacks[GiftType.KillsForceCrits]) < Time.time) return;
+            critInfo.guaranteed = true;
+        }
+
+        static void RegisterPurchase(EntityBoughtInfo info)
+        {
+            CurrentPurchases++;
+        }
+
+        // GiftType.CratesDropMoreItems
+        static void SpawnMoreCrateStuff(KillInfo info)
+        {
+            if (info == null) return;
+            CrateContainsStuff component = info.killHit.target.GetComponent<CrateContainsStuff>();
+            if (component == null) return;
+            component.SpawnMore((int) Math.Pow(2, CurrentStacks[GiftType.SpeedAndKnockbackIncrease] + 1) - 1);
+        }
+    }
+
+    class GiftEffects
+    {
+        static readonly string[] BannedFollowers = {
+            "CubicleGroundBotElectricity Variant",
+            "CubicleGroundBotFire Variant",
+            "CubicleGroundBotIce Variant",
+            "CubicleGroundBotSawblade Variant",
+            "CubicleGroundBotSmall",
+            // Spawns with no can ):
+            "JoblinCanbassador"
+        };
+
+        public static bool RunGiftOnApplied(GiftType giftType, BackupSkill skillInstance)
+        {
+            CutterConfig cornercutter = CutterConfig.Instance;
+            DebugStuff resources = cornercutter.GetDebugResources();
+            if (resources == null) return false;
+            switch (giftType)
+            {
+                case GiftType.RandomSkills:
+                    for (int i = 0; i < 5; i++)
+                    {
+                        resources.GainRandomSkill();
+                    }
+                    break;
+                case GiftType.RandomFollowers:
+                    for (int i = 0; i < 5; i++)
+                    {
+                        GameObject companion = resources.AllEnemies.RandomSelection((GameObject f) => 1);
+                        if (Array.IndexOf(BannedFollowers, companion.name) != -1)
+                        {
+                            i--;
+                            continue;
+                        }
+                        cornercutter.LogDebug("Selected companion as " + companion.name);
+                        EnemyCompanion.AddBuff(skillInstance.moddedEntity, companion, skillInstance.moddedEntity.transform.position, EnemyCompanion.CompanionType.Platonic, false);
+                    }
+                    break;
+                case GiftType.SpeedAndKnockbackIncrease:
+                    Entity ent = skillInstance.moddedEntity;
+                    if (ent is Player)
+                    {
+                        Player player = ent as Player;

```

Comment:
I think this (And the usage in the method below) can be replaced with `if (ent is Player player)`